visma/simplify/simplify.py

```diff
@@ -165,6 +167,97 @@ def simplifyEquation(lToks, rToks):
 
 
 def simplify(tokens):
+    tokens_orig = copy.deepcopy(tokens)
+    animation = [tokens_orig]
+    comments = [[]]
+    tokens, availableOperations, token_string, anim1, comment1 = expressionSimplification(tokens_orig, [], tokens)
+    animation.extend(anim1)
+    comments.extend(comment1)
+    return tokens, availableOperations, token_string, animation, comments
+
+
+def expressionSimplification(tokens_now, scope, tokens1):
+    animation = []
+    comments = []
+    simToks = []
+    mulFlag = True
+    expresstionMultiplication = False
+    for _ in range(50):
+        for i, _ in enumerate(tokens1):
+            mulFlag = False
+            if isinstance(tokens1[i], Expression):
+                if (i > 1):
+                    if (tokens1[i - 1].value == '*'):
+                        scope.append(i)
+                        tokens1[i].tokens, _, _, _, _ = expressionSimplification(tokens_now, scope, tokens1[i].tokens)
+                        if isinstance(tokens1[i - 2], Expression):
+                            scope.append(i - 2)
+                            tokens1[i - 2].tokens, _, _, _, _ = expressionSimplification(tokens_now, scope, tokens1[i - 2].tokens)
+                        a = tokens1[i - 2]
+                        b = tokens1[i]
+                        c = a * b
+                        mulFlag = True
+                        expresstionMultiplication = True
+                        if isinstance(c, Expression):
+                            scope.append(i)
+                            c.tokens, _, _, _, _ = expressionSimplification(tokens_now, scope, c.tokens)
+                        tokens1[i] = c
+                        del tokens1[i - 1]
+                        del tokens1[i - 2]
+                        break
+        if not mulFlag:
+            break
+    if expresstionMultiplication:
+        animation.append(tokens1)
+        comments.append(['Multiplying expressions'])
```

Comment:
This currently doesn't show which expressions are being multiplied and all the expressions multiplications are simplified at once.
For now, add a `TODO`.
```suggestion
        comments.append(['Multiplying expressions'])
        # TODO: Implement verbose multiplication steps
```